### PR TITLE
Group Dependencies in Renovate

### PR DESCRIPTION
**Pull request by manjunathb4461**
Started looking into this after discussions [here](https://chainlink-core.slack.com/archives/C03UV9B7M3L/p1719583508860899?thread_ts=1719571377.932819&cid=C03UV9B7M3L) 

- Renovate will group all minor and patch dependencies together by package manager name and create PRs with branch named after package manager `npm-dependencies`, `github-actions-dependencies` etc
- Major dependencies will continue to be raised separately per dependency

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -200,6 +200,8 @@ importers:
 
   actions/nops-upgrade-checker: {}
 
+  actions/policy-bot-config-validator: {}
```
Comment (manjunathb4461) [inline pnpm-lock.yaml:203]: added by lefthook

File: .github/renovate.json [hunk 2]
```diff
@@ -21,9 +21,14 @@
   },
   "packageRules": [
     {
-      "matchDepTypes": ["devDependencies"],
-      "matchUpdateTypes": ["patch"],
-      "enabled": false
+      "groupName": "major-minor dependencies grouped by manager",
+      "branchName": "{{manager}}-dependencies",
+      "matchUpdateTypes": ["major", "minor"]
```
Comment (momentmaker) [inline .github/renovate.json:26]: it might make sense to not have `major` to be grouped as it probably has breaking changes

Comment (manjunathb4461) [reply .github/renovate.json:26]: Should we group all minor and patch dependencies together per package manager and major updates can continue separately?

Comment (momentmaker) [reply .github/renovate.json:26]: yeah that sounds good :)

Comment (manjunathb4461) [reply .github/renovate.json]: updated accordingly, PTAL

**Review by momentmaker — APPROVED** [review]

**Pull request merged**